app/frontend/src/pages/ask/Ask.tsx

```diff
@@ -48,13 +48,15 @@ export function Component(): JSX.Element {
     const [showSpeechInput, setShowSpeechInput] = useState<boolean>(false);
     const [showSpeechOutputBrowser, setShowSpeechOutputBrowser] = useState<boolean>(false);
     const [showSpeechOutputAzure, setShowSpeechOutputAzure] = useState<boolean>(false);
+    const audio = useRef(new Audio()).current;
+    const [isPlaying, setIsPlaying] = useState(false);
 
     const lastQuestionRef = useRef<string>("");
 
     const [isLoading, setIsLoading] = useState<boolean>(false);
     const [error, setError] = useState<unknown>();
     const [answer, setAnswer] = useState<ChatAppResponse>();
-    const [speechUrl, setSpeechUrl] = useState<string | null>(null);
+    const [speechUrl, setSpeechUrl] = useState<(string | null)[]>([]);
```

Comment:
Since this is a list, can you change to speechUrls, setSpeechUrls, etc?

Reply:
Even though it's a list, it consists of only one url, wasn't sure about naming it like that but I'm happy to make the change.

Reply:
Ah, is it only a list so that you can update it from the child component?
Is this the standard approach in React these days for child->parent updates?

Reply:
Not sure if it's the standard or not.
But the Chat.tsx requires it to be a list as there are a list of messages,
while the Ask.tsx doesn't require it to be a list as there is only one message.

The child component needs it be of the same type as it's managed using a state to update the Parent (i.e. speechUrls in Chat.tsx) state from the Child (i.e. SpeechAzure)
So, I opted into creating it as a list for both, keep in mind that this is needed to always have a list of generated urls to not regenerate them every time.

All of this weird logic can be removed if the update was done in the Chat.tsx but to keep it modular all of these stuff are required to be passed around and have the same types.


If you have a better idea / prefer something else please let me know.

Reply:
I recommend you take another look on it, I'm not as reluctant as I was about now that I added the SpeechConfig thingy